content_script.js

```diff
@@ -134,6 +89,91 @@ class HTMLSerializer {
   }
 
   /**
+   * Takes an html element, and populates this object's fields with the
+   * appropriate attribute names and values.
+   *
+   * @param {Element} element The Element to serialize.
+   * @private
+   */ 
+  processAttributes(element) {
+    var win = element.ownerDocument.defaultView;
+    var style = win.getComputedStyle(element, null).cssText;
+
+    // TODO(sfine): Ensure that getComputedStyle returns absolute urls.
+    //              https://www.w3.org/TR/css3-values/ 3.4.1
+
+    var windowDepth = this.windowDepth(window);
+    style = style.replace(/"/g, this.escapedQuote(windowDepth+1));
+    var quote = this.escapedQuote(windowDepth);
+    this.html.push(`style=${quote}${style}${quote} `);
+
+    var attributes = element.attributes;
+    if (attributes) {
+      for (var i = 0, attribute; attribute = attributes[i]; i++) {
+        switch (attribute.name.toLowerCase())  {
+          case 'src':
+            this.processSrcAttribute(element);
+          case 'style':
+            break;
+          default:
+            var name = attribute.name;
+            var value = attribute.value;
+            this.addSimpleAttribute(name, value);
+        }
+      }
+      // TODO(sfine): Ensure this is working by making sure that an iframe
+      //              will always have attributes.
+      if (element.tagName.toLowerCase() == 'iframe') {
+        this.html.push('srcdoc=');
+        var name = this.iframeFullyQualifiedName(element.contentWindow);
+        this.frameHoles[this.html.length] = name;
+        this.html.push(''); // Entry where the iframe contents will go.
+      }
+    }
+  }
+
+  /**
+   * Process the src attribute of a given element.
+   *
+   * @param {Element} element The element being processed, which has the src
+   *     attribute.
+   * @private
+   */
+  processSrcAttribute(element) {
+    switch(element.tagName.toLowerCase()) {
+      default:
```

Comment:
The purpose of "case 'iframe'" is to simply do nothing.  Should I have put it before the default with a break statement?